### DISCUSSION TITLE
Lesson 7: pushing to Github: git status

### DISCUSSION THREAD

**Discussion by Northh27** [Cyfrin/foundry-full-course-cu#967] [Q&A]
Hello
When I was watching the video and seeing the files that were going to get pushed up to github. I saw some of my desktop folders in the file. That is because I had my Foundry-f23 folder in a desktop folder. So basically my desktop folders were getting pushed up. I was able to change what file/path foundry-f23 was into an empty folder so it's the only file getting pushed up.

The problem now is that when i do "git status" for the folder i have some missing files compared to Patrick. This is at 2:27:15 in the video

```
On branch main

No commits yet

Changes to be committed:
  (use "git rm --cached <file>..." to unstage)
        new file:   .gas-snapshot
        new file:   .github/workflows/test.yml
        new file:   .gitignore
        new file:   Makefile
        new file:   README.md
        new file:   foundry.toml
        new file:   script/DeployFundMe.s.sol
        new file:   script/HelperConfig.s.sol
        new file:   script/Interactions.s.sol
        new file:   src/FundMe.sol
        new file:   src/PriceConverter.sol
        new file:   test/Interaction/InteractionTest.t.sol
        new file:   test/Mocks/MockV3Aggregator.sol
        new file:   test/unit/FundMeTest.t.sol
        ```
        Also when I do git log I get
        ```
        fatal: your current branch 'main' does not have any commits yet
        ```
        Can someone please explain how i can fix this?

**Comment by Northh27** [comment]
This is suppose to be on its own

Also when I do git log I get
        ```
        fatal: your current branch 'main' does not have any commits yet
        ```
        Can someone please explain how i can fix this?

**Reply by theirrationalone** [reply]
Heyy @Northh27 !, You may missing some steps to push your codebase into your repo...

Recommended steps to push codebase on github:

After creating a repo on github, Please copy the repo's url and bring it into your terminal.

such as here... (text after // all are comments inclusively).
```bash
git remote add origin 'your_git_repo_link' // don't include both ', This cmd helps you to set the remote repo as your codebase origin, however you can change your origin name from origin to anything you wish. Your chosen name then will be your repo's origin, it is considered as best practice to name your repo's origin as origin and nothing special about it.

git remote -v // this cmd helps you to check your remote origin (repo url).

git add . // this cmd makes files staged.

git commit -m "your commit msg" // this cmd helps you to make a commit on staged files.

git fetch origin // this cmd helps you to sync with your remote repo latest state.

git push -u origin main // this cmd helps you to push your commits on to the repo.
```

**Comment by Jsyn00** [comment]
That fatal error, when using the "git log" command, means that you have not made any commits yet. In Patrick's video 2 commits appear because he uses another version of git, but they don't have to appear to you.

If you continue with the video and make the commit as Patrick explains, then when you do "git log" your first commit will appear correctly
